### PR TITLE
legacy: Add settings for coredns to fix problems with custom cluster ip ranges

### PR DESCRIPTION
**Pull request by rossf7**
Same as https://github.com/giantswarm/cluster-operator/pull/871 but for legacy.

### CODE REVIEW

File: pkg/project/project.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package project
 
 var (
-	bundleVersion = "0.21.1-dev"
+	bundleVersion = "0.21.1-dev-rossf7"
```
Comment (rossf7) [inline pkg/project/project.go]: TODO

File: service/controller/aws/resource_set.go [hunk 2]
```diff
@@ -199,8 +199,10 @@ func newResourceSet(config resourceSetConfig) (*controller.ResourceSet, error) {
 			K8sClient:                config.K8sClient.K8sClient(),
 			Logger:                   config.Logger,
 
-			ClusterIPRange: config.ClusterIPRange,
-			Provider:       config.Provider,
+			CalicoAddress:      config.CalicoAddress,
+			CalicoPrefixLength: config.CalicoPrefixLength,
```
Comment (tomahawk28) [inline service/controller/aws/resource_set.go]: PR is about coredns, but only I can see these two additional variables, is it okay?

**Comment by rossf7** [comment]
Tested on `gorgoth`.

```
$ helmgs get values coredns

baseDomain: h9k4m.k8s.gorgoth.gridscale.kvm.gigantic.io
cluster:
  calico:
    CIDR: 172.24.0.0/16
  kubernetes:
    API:
      clusterIPRange: 172.31.0.0/16
    DNS:
      IP: 172.31.0.10
clusterDNSIP: 172.31.0.10
clusterID: h9k4m
exporter:
  namespace: kube-system
externalDNSIP: 8.8.8.8
image:
  registry: quay.io
tiller:
  namespace: giantswarm
```

File: service/controller/aws/resource_set.go (hunk 2 above)
Comment (rossf7) [reply service/controller/aws/resource_set.go]: coredns needs to know the Calico CIDR. This is the values used now.

```
clusterDNSIP: 172.31.0.10
```

And after the fix.

```
cluster:
  calico:
    CIDR: 172.24.0.0/16
  kubernetes:
    API:
      clusterIPRange: 172.31.0.0/16
    DNS:
      IP: 172.31.0.10
clusterDNSIP: 172.31.0.10
```

**Comment by MarcelMue** [comment]
Can we get this merged? I would like to test it for the Azure 1.16 release.

**Comment by rossf7** [comment]
MarcelMue Ack.

Tested on `gorgoth`. Going with this.

**Pull request merged**